src/server/qgsserverrequest.cpp

```diff
@@ -20,6 +20,17 @@
 #include "qgsserverrequest.h"
 #include <QUrlQuery>
 
+
+const QString QgsServerRequest::HOST = QStringLiteral( "Host" );
+const QString QgsServerRequest::FORWARDED = QStringLiteral( "Forwarded" );
+const QString QgsServerRequest::X_FORWARDED_HOST = QStringLiteral( "X-Forwarded-Host" );
+const QString QgsServerRequest::X_FORWARDED_PROTO = QStringLiteral( "X-Forwarded-Proto" );
+const QString QgsServerRequest::X_QGIS_SERVICE_URL = QStringLiteral( "X-Qgis-Service-Url" );
+const QString QgsServerRequest::X_QGIS_WMS_SERVICE_URL = QStringLiteral( "X-Qgis-Wms-Service-Url" );
+const QString QgsServerRequest::X_QGIS_WFS_SERVICE_URL = QStringLiteral( "X-Qgis-Wfs-Service-Url" );
+const QString QgsServerRequest::X_QGIS_WCS_SERVICE_URL = QStringLiteral( "X-Qgis-Wcs-Service-Url" );
+const QString QgsServerRequest::X_QGIS_WMTS_SERVICE_URL = QStringLiteral( "X-Qgis-Wmts-Service-Url" );
```

Comment:
A Q_ENUM would probably be more elegant.

Reply:
I don't catch what better with a Q_ENUM, it's just used Headers names...

Reply:
I think it's more readable and robust to rely on enum instead of plain string.

Reply:
I think that an enum and a static const dictionary would be more readable than polluting the `QgsServerRequest` class.

Something that you can call like `request->header(QgsServerRequest::RequestHeader::X_FORWARDED_HOST)`.